### PR TITLE
R packages

### LINKED ISSUE #5601: Adding a dialogue to install a new (or updated?) R package
This is (sort of) just the install.packages("xxx") command and I use this quite often from the script window.
Could we have a simple dialogue to do this.  It could be a new Packages tab in the Tools > Options dialogue.
The R code isn't trivial because it would be good if:
a) It checks first if the package has already been installed.
b) It checks if the package exists - could be a typing error in the name of the package.
This is discussed in stack-overflow for example [here](https://stackoverflow.com/questions/9341635/check-for-installed-packages-before-running-install-packages).

### PR DESCRIPTION
**Pull request by Ivanluv**
@africanmathsinitiative/developers  This is ready for review fixes #5601

### CODE REVIEW

**Comment by Ivanluv** [comment]
@rdstern would you kindly look at this again

**Comment by rdstern** [comment]
ok  Thanks

**Comment by rdstern** [comment]
Looked again and works very smoothly now - when it loads.  It no longer asks for the mirror or the package.
When I make a mistake, e.g. try just putting a space into the package name - or (say) xxx, then the check says there is no package name.  But OK is enabled.  It doesn't give an error and the command line - in the output window is just the same.  

I queried before whether it could still give the list of packages available - which it did before, even when the name was ok.  This time I would like it to give that list if there is an error, so we can see what is available?  Is that possible?

I would like it to do something different when it is successful compared to when it fails.  Can it say anything about the loaded system in the output window?  Whan you give the command in the script window that information is displayed  in its own window, though not in the output.

**Comment by Ivanluv** [comment]
`install.package`  does not give an error when the package is wrong or does not exist.It however gives a warning which is capture correctly in Rstudio but not captured in the R Console.An option is upgrading a warning to o an error using `option(warn=2)` which adds an extra warning

File: instat/dlgRPackages.vb [hunk 1]
```diff
@@ -0,0 +1,101 @@
+﻿Imports instat
```
Comment (shadrackkibet) [inline instat/dlgRPackages.vb]: This is not needed

File: instat/dlgRPackages.vb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        ucrInputTextBoxRPackage.SetParameter(New RParameter("pkgs", 1))
+        CheckEnable()
+    End Sub
+
+    Private Sub SetDefaults()
+        clsInstallPackage = New RFunction
+        clsInstallPackage.SetRCommand("install.packages")
+        clsInstallPackage.AddParameter("repos", Chr(34) & "https://cran.rstudio.com/" & Chr(34))
+        ucrBase.clsRsyntax.SetBaseRFunction(clsInstallPackage)
+    End Sub
```
Comment (shadrackkibet) [inline instat/dlgRPackages.vb]: A space after each sub.

File: instat/dlgRPackages.vb [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    Private Sub InitialiseDialog()
+        ucrBase.iHelpTopicID = 592
+        ucrInputTextBoxRPackage.SetParameter(New RParameter("pkgs", 1))
+        CheckEnable()
+    End Sub
+
+    Private Sub SetDefaults()
+        clsInstallPackage = New RFunction
+        clsInstallPackage.SetRCommand("install.packages")
+        clsInstallPackage.AddParameter("repos", Chr(34) & "https://cran.rstudio.com/" & Chr(34))
```
Comment (shadrackkibet) [inline instat/dlgRPackages.vb]: Parameter position?

File: instat/dlgRPackages.vb [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    End Sub
+
+    Private Sub ucrInputTextBoxRPackage_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrInputTextBoxRPackage.ControlContentsChanged
+        ucrInputMessage.SetText("")
+        ucrInputMessage.txtInput.BackColor = Color.White
+        CheckEnable()
+        TestOkEnabled()
+    End Sub
+
+    Private Sub check()
```
Comment (shadrackkibet) [inline instat/dlgRPackages.vb]: capital "C"

File: instat/dlgRPackages.vb [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+    Private Sub CheckEnable()
+        If Not ucrInputTextBoxRPackage.IsEmpty Then
+            cmdCheck.Enabled = True
+            ucrInputMessage.Enabled = True
+        Else
+            cmdCheck.Enabled = False
+            ucrInputMessage.Enabled = False
+        End If
+    End Sub
```
Comment (shadrackkibet) [inline instat/dlgRPackages.vb]: I think it is possible to do this through our linking system.

File: instat/dlgRPackages.vb [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+            InitialiseDialog()
+            bFirstLoad = False
+        End If
+        If bReset Then
+            SetDefaults()
+        End If
+        SetRCodeForControls(bReset)
+        bReset = False
+        TestOkEnabled()
+    End Sub
```
Comment (shadrackkibet) [inline instat/dlgRPackages.vb]: space

**Comment by maxwellfundi** [comment]
@Ivanluv  this needs a few changes to get it ready to merge. Could you get these few changes done soon?

**Comment by maxwellfundi** [comment]
@rdstern Please review this.

**Comment by maxwellfundi** [comment]
Tested this with @rdstern :-)

**Pull request merged**